insights/parsers/krb5.py

```diff
@@ -0,0 +1,108 @@
+    "false"
+    >>> krb5_info.data["realms"]["EXAMPLE.COM"]["kdc"]
+    "kerberos.example.com"
+"""
+
+
+from insights.core import Parser
+from insights.core.plugins import parser
+
+
+@parser('krb5')
+@parser('krb5.conf.d')
+class Krb5Configuration(Parser):
+    """Class for krb5.conf and krb5.conf.d configuration."""
+
+    def parse_content(self, content):
+        dict_all = {}
+        is_squ = False
+        section_name = ""
+        squ_value = {}
+        squ_section_name = ""
+        section_value = {}
+        for line in content:
+            """Get active line """
+            if "includedir" in line or line.startswith("#") or not line:
+                continue
+            """[XXX] means a section, get the section name and the value format is dict."""
+            if "[" in line and "]" in line:
+                """If first section, just get the section name, if not, set the value to the former section and get the section name"""
+                if not section_name:
+                    section_name = line.strip().strip("[]")
+                    section_value = {}
+                else:
+                    dict_all[section_name] = section_value
+                    section_name = line.strip().strip("[]")
+                    section_value = {}
+                continue
+            """Common key value format is XXX = YYY"""
+            if "=" in line and "{" not in line and "}" not in line and not is_squ:
+                key_value = line.strip().split("=")
+                section_value[key_value[0].strip()] = key_value[1].strip()
+                continue
+            """"Special key value format is XXX = YYY{%ZZZ}"""
+            if "=" in line and "{" in line and "}" in line and len(line.split("=")) == 2 and not is_squ:
+                key_value = line.strip().split("=")
+                section_value[key_value[0].strip()] = key_value[1].strip()
+                continue
+            """If in {} sub_section, get the key_value pair"""
+            if "=" in line and is_squ:
+                key_value = line.strip().split("=")
+                squ_value[key_value[0].strip()] = key_value[1].strip()
+                continue
+            """The {} sub_section should start with format XXX = {"""
+            if "=" in line and "{" in line and "}" not in line and len(line.split("=")) == 2:
+                is_squ = True
+                squ_section_name = line.strip().split("=")[0].strip()
+                continue
+            """The {} sub_section should end with }, if it is, set the whole value to the sub_section name, and clean the flag"""
+            if "}" in line and "=" not in line and is_squ:
+                section_value[squ_section_name] = squ_value
+                is_squ = False
+                squ_section_name = ""
+                squ_value = {}
+                continue
+        """Set the value to the last section """
+        dict_all[section_name] = section_value
+        self.data = dict_all
```

Comment:
It seems you have looked at getting the contents of this file, but you have not considered how a rule writer would use the contents.  How does a rule writer find out what realms are defined?  Can they easily find out the ticket lifetime for any domain, with reference back to the defaults?  How would a rule get the reverse mapping from realm back to its domains?

As well as the usual IniConfigFile mechanisms, I would recommend you think about how best to provide useful information from the file to save future rule writers effort.